language/move-lang/src/hlir/translate.rs

```diff
@@ -1317,7 +1318,9 @@ fn bind_for_short_circuit(e: &T::Exp) -> bool {
     use T::UnannotatedExp_ as TE;
     match &e.exp.value {
         TE::Use(_) | TE::InferredNum(_) => panic!("ICE should have been expanded"),
-        TE::Value(_) | TE::Move { .. } | TE::Copy { .. } | TE::UnresolvedError => false,
+        TE::Value(_) | TE::Move { .. } | TE::Copy { .. } | TE::Spec(_) | TE::UnresolvedError => {
```

Comment:
Move this `TE::Spec {}` into the same spot as `Unit` below

Reply:
Done.